ZenPacks/zenoss/Microsoft/Windows/WinService.py

```diff
@@ -161,13 +157,71 @@ def get_winservices_modes(self):
         ''''''
         data = {}
         for svc in self.device().os.winservices():
-            if svc.monitored():
+            if svc.monitor:
                 data[svc.id] = {'modes': svc.get_serviceclass_startmodes(),
                                 'mode': svc.startMode,
-                                'monitor': svc.monitored()
+                                'monitor': svc.monitor,
+                                'severity': svc.getFailSeverity(),
                                 }
         return data
 
+    def getMonitoredDataSource(self):
+        '''Return datasource for template if it exists'''
+        # first there must be a template
+        if not self.datasource_id:
+            self.setMonitoredDataSource()
+        # return the datasource if it is set
+        if self.datasource_id:
+            # return monitored datasource
+            template = self.getRRDTemplate()
+            if template:
+                datasource = template.datasources._getOb(self.datasource_id, None)
+                if datasource and self.getMonitored(datasource):
+                    return datasource
+        # if returning the datasource fails reset the datasource_id to none
+        self.datasource_id = None
+        return None
+
+    def setMonitoredDataSource(self):
+        ''' set the datasource_id parameter if
+            a valid datasource can be found
+        '''
+        def test_datasource(ds):
+            '''set self.datasource_id if valid'''
+            if hasattr(ds, 'startmode') and self.getMonitored(ds):
+                self.datasource_id = ds.id
+
+        # if not defined, try to define it
+        if not self.datasource_id:
+            template = self.getRRDTemplate()
+            if template:
+                # first check DefaultService
+                datasource = template.datasources._getOb('DefaultService', None)
+                if datasource:
+                    test_datasource(datasource)
+                # if it's still undefined, check for other datasources
+                if not self.datasource_id:
+                    for datasource in template.getRRDDataSources():
+                        if datasource.id == 'DefaultService': continue
+                        test_datasource(datasource)
+
+    def getFailSeverity(self):
+        """
+        Return the severity for this service when it fails.
+        """
+        template = self.getRRDTemplate()
+        if template:
+            datasource = template.datasources._getOb(self.datasource_id, None)
+        if datasource:
```

Comment:
probably wouldn't happen, but if template wasn't found, then datasource would not be set and we'd get an exception.  probably best to just indent these 2 lines